### DISCUSSION TITLE
After changing a `RwSignal` from an OS callback, the UI is not updated until focused/hovered

### DISCUSSION THREAD

**Discussion by kaivol** [lapce/floem#451] [General]
On Windows, I'm using a `RwSignal` to store (and show) the system's current accent color. 
Further, I register a `SUBCLASSPROC` callback using `SetWindowSubclass` to get notified when the accent color is changed: In the callback, I update the value in the `RwSignal`.  

If I now change the accent color in the Windows settings the callback is invoked immediately, but the UI does not show this change until I hover the window with the mouse or focus the window. 
A workaround that I have found is calling 
````rust
floem::ext_event::register_ext_trigger(Trigger::new());
````
after changing the `RwSignal`. This causes the UI to be updated immediately. 

Is this the expected/desired behavior, and if so, what would be the best way to solve the problem?

Thanks!

**Comment by panekj** [comment]
That's because almost every Windows API is async. This is the expected behaviour.

**Reply by kaivol** [reply]
Could you elaborate on what you mean by “async” in this context?
As far as I understand it, `SetWindowSubclass` registers an ordinary callback which gets invoked in the main thread.
